Add a waitevent column to blocking queries

blogh opened the pull request:
Blocking queries can be long for a number of reasons, the wait_event
column of pg_activity can help us understand them. Therefore, it was
added to the blocking query panel.

blogh commented:
This PR is still lacking the man page update.
I'll wait for #216  to be merged before adding it.

blogh commented:
Tests would also have to be added when #217 is merged.

blogh commented:
Man page updated

dlax commented on pgactivity/types.py:
```diff
@@ -702,7 +703,7 @@ class RunningProcess(BaseProcess):
 @attr.s(auto_attribs=True, frozen=True, slots=True)
-class BWProcess(BaseProcess):
+class WaitingProcess(BaseProcess):
     """Process for a blocking or waiting query."""
```
```suggestion
    """Process for a waiting query."""
```

dlax commented on pgactivity/queries/get_blocking.sql:
```diff
@@ -18,7 +18,8 @@ SELECT
           ELSE sq.query
-      END AS query
+      END AS query,
+      NULL AS wait
```
Why not using `pg_stat_activity.waiting`?

blogh replied:
no reason .. will fix that way

dlax commented on docs/man/pg_activity.pod:
```diff
@@ -139,6 +139,8 @@ requiered by another session. It shows following information:
 
 =item - B<TIME+>: the duration of the query, transaction or session depending on the B<DURATION_MODE> setting;
 
+=item - B<Waiting>: specific wait event if available. Otherwise, a boolean indicating it we are waiting for a Lock;
```
```suggestion
=item - B<Waiting>: specific wait event if available. Otherwise, a boolean indicating if we are waiting for a Lock;
```

(Besides, it's not quite accurate that "we are waiting for a Lock" and it's not always a boolean either. It's unrelated to this PR, but we might want to update this afterwards.)

blogh commented:
Fixed the mistakes and tried to reword the description of the waiting column in the man page

dlax commented on pgactivity/data.py:
```diff
@@ -278,13 +278,15 @@ def pg_get_waiting(self, duration_mode: int = 1) -> List[BWProcess]:
+            query = queries.get("get_blocking_post_100000")
+        elif self.pg_num_version >= 90200:
             query = queries.get("get_blocking_post_90200")
         elif self.pg_num_version < 90200:
```
For clarity, I think we should turn this `elif` into an `else`.

blogh replied:
You mean the last one ? if so should I fix the other instances of this in data.py ?

dlax replied:
Yes. If there are more of these, I'd suggest to clean this up in a dedicated commit as this is unrelated to this change.

blogh replied:
done

dlax reviewed (changes requested):
One minor nit, LGTM otherwise.

blogh commented:
* Fixed the if .. elif ..else
* Fixed a mystake on my part wait_event where introduced in 9.6 not 10 (But only for lock / lwlocks / bufferpin)
  * modified the manpage
  * modified version checks for pg9.6 and query file
  
 Test pass from 9.6 to 13.
